### DISCUSSION TITLE
Why bSendRpc for replicated property change?

### DISCUSSION THREAD

**Discussion by watsonsong** [Sixze/ALS-Refactored#532] [Q&A]
In methods like [SetViewMode](https://github.com/Sixze/ALS-Refactored/blob/44227890452530f97f29244ae4d1bd7855bef24f/Source/ALS/Private/AlsCharacter.cpp#L448), the bSendRpc default is true. When on server side mark property dirty, it still send rpc to client. Is is unnesessary because the property will replicated to client when it mark dirty. And the order about property replicate and RPC may not in sequence, maybe it will has some error result?

**Comment by ameaninglessname** [comment]
The view mode is set to "skip owner", so it won't be replicated to the owning client. Therefore, they need to communicate with RPC.

**Reply by watsonsong** [reply]
I understand. So this pattern is used to optimization if I modify view mode on the owner side?

**Reply by ameaninglessname** [reply]
It's a optimization for owning client to not replicated `view mode` to it, because we've done it through RPC.

The pattern used here, IMO, is `client(-or-server)-driven", similar to what `UCharacterMovementComponent` does.
